types/index.d.ts

```diff
@@ -836,7 +836,23 @@ declare module 'cloudinary' {
         value: string;
     }
 
-    export type AnalysisType = 'captioning' | 'cld_fashion' | 'cld_text' | 'coco' | 'google_tagging' | 'human_anatomy' | 'lvis' | 'shop_classifier' | 'unidet' | 'custom';
+    export type AnalysisType = CustomizableAnalysisType | NonCustomizableAnalysisType;
+
+    export type CustomizableAnalysisType = 'custom';
```

Comment:
Please assume that all analysis types will be customizable (i.e. accept parameters)
It's just not the case right now because the API is not very mature.

Also, as mentioned before, some parameters will have the same name. For example, soon `model_version` can be part of `captioning` as well, so both of these will be valid:

```
{
	"analysis_type": "captioning",
    "parameters":
    {
        "captioning":
        {
            "model_version": 4
        }
    }
}
```

```
{
	"analysis_type": "custom",
    "parameters":
    {
        "custom":
        {
        	"model_name": "my-model",
            "model_version": 4
        }
    }
}
```